### DISCUSSION TITLE
[Feature Request] Ability to Effect how folders and tests are shown rather than clog the page

### DISCUSSION THREAD

**Discussion by ApiRulez** [DannyDainton/newman-reporter-htmlextra#363] [Ideas]
**Is your feature request related to a problem? Please describe.**
Yes, when i have a large collection that has folders, although it is nice for the report to show all folders, would be Wouldit it hid the subfolders.

**Describe the solution you'd like**
When you have a Collection that has a folder structure like:

MAIN
  - Test Set 1
      - Test 1
      - Test 2
      - Test 3
   - Test Set 2
      - Test 1
      - Test 2

Rather than show a HTML Report that shows:

MAIN / Test Set 1 / Test 1 - 1 REQUEST IN THE FOLDER
MAIN / Test Set 1 / Test 2 - 1 REQUEST IN THE FOLDER
MAIN / Test Set 1 / Test 3 - 1 REQUEST IN THE FOLDER

MAIN / Test Set 2 / Test 1 - 1 REQUEST IN THE FOLDER
MAIN / Test Set 2 / Test 2 - 1 REQUEST IN THE FOLDER

Have a switch that allows them to just show as:

MAIN / Test Set 1 - 3 REQUEST IN THE FOLDER
MAIN / Test Set 2 - 3 REQUEST IN THE FOLDER

and then they can be expanded to show the folders underneath, so it looks more like it does inside Postman

**Comment by DannyDainton** [comment]
Hey @ApiRulez 

The report is created from a template (https://github.com/DannyDainton/newman-reporter-htmlextra/blob/master/lib/dashboard-template.hbs) you can copy and change any of that to suit what you specifically want to see.

Using the `--reporter-htmlextra-template <path to file>` flag will allow you to use that for your Newman runs.

**Comment by ApiRulez** [comment]
Hey @DannyDainton, thanks for the quick response :)  I did have a look at this, but it's a bit beyond my comprehension. Will try and have ago now though.

**Comment by DannyDainton** [comment]
If I get some spare time this week I might have a look but I can't really promise anything :D

**Comment by ApiRulez** [comment]
Thanks @DannyDainton

**Comment by DannyDainton** [comment]
Not had a chance to look at this and probably won't anytime soon.

All PRs are welcome if someone wants to give it a go.
